bfcfg

```diff
@@ -648,29 +655,157 @@ sys_cfg_one_byte()
   fi
 }
 
+#
+# Dump a multi-byte value at the specified offset in a file.
+#
+sys_cfg_multi_byte()
+{
+  local tmp_file=$1
+  local name=$2
+  local offset=$3
+  local num_bytes=$4
+  local value=$5
+
+  if [ ${dump_mode} -eq 0 ]; then
+    return
+  fi
+
+  value=0
+  if [ -e ${tmp_file} ]; then
+    value=$(hexdump -s "${offset}" -n "${num_bytes}" -e '"0x%X" "\n"' "${tmp_file}")
+  fi
+  echo "sys: ${name}=${value}"
+}
+
+#
+# This function reads the BfModes EFI struct from a UEFI variable directly.
+# Below are the offsets defined in UEFI which is 4(fixed header) plus the offset
+# of the variable within the BfModes struct. These offsets are not supposed to
+# change in order to be backward compatible with previous releases.
+#   VARIABLE(Name)       OFFSET(Byte)  SIZE(Byte)
+#   INTERNAL_CPU_MODEL        4            1
+#   HOST_PRIV_LEVEL           5            1
+#   NIC_MODE                  6            1
+#
+sys_cfg_dump_bf_modes()
+{
+  local tmp_file=/tmp/.bfcfg-sysfs-data
+  local internal_cpu_model internal_cpu_model_str
+  local host_priv_level host_priv_level_str
+  local nic_mode nic_mode_str
+
+  if [ ${dump_mode} -eq 0 ]; then
+    return
+  fi
+
+  if [ ! -e "${bf_modes_sysfs}" ]; then
+    log_msg "sys: failed to find the ${bf_modes_sysfs} EFI variable"
+    return
+  fi
+
+  # shellcheck disable=SC2216
+  yes | cp -f ${bf_modes_sysfs} ${tmp_file}
+  has_change=0
+
+  # shellcheck disable=SC2216
+  yes | cp -f ${bf_modes_sysfs} ${tmp_file}
+  internal_cpu_model=$(hexdump -s 4 -n 1 -e '/1 "%d" "\n"' ${tmp_file})
+
+  # shellcheck disable=SC2216
+  yes | cp -f ${bf_modes_sysfs} ${tmp_file}
+  host_priv_level=$(hexdump -s 5 -n 1 -e '/1 "%d" "\n"' ${tmp_file})
+
+  # shellcheck disable=SC2216
+  yes | cp -f ${bf_modes_sysfs} ${tmp_file}
+  nic_mode=$(hexdump -s 6 -n 1 -e '/1 "%d" "\n"' ${tmp_file})
+
+  if [ "${internal_cpu_model}" -eq 0 ]; then
+    internal_cpu_model_str="SEPARATED"
+  elif [ "${internal_cpu_model}" -eq 1 ]; then
+    internal_cpu_model_str="EMBEDDED"
+  else
+    internal_cpu_model_str="UNKNOWN"
+  fi
+
+  if [ "${host_priv_level}" -eq 0 ]; then
+    host_priv_level_str="PRIVILEGED"
+  elif [ "${host_priv_level}" -eq 1 ]; then
+    host_priv_level_str="RESTRICTED"
+  else
+    host_priv_level_str="UNKNOWN"
+  fi
+
+  if [ "${nic_mode}" -eq 0 ]; then
+    nic_mode_str="DPU_MODE"
+  elif [ "${nic_mode}" -eq 1 ]; then
+    nic_mode_str="NIC_MODE"
+  else
+    nic_mode_str="UNKNOWN"
+  fi
+
+  echo "sys: INTERNAL_CPU_MODEL=${internal_cpu_model_str}"
+  echo "sys: HOST_PRIV_LEVEL=${host_priv_level_str}"
+  echo "sys: NIC_MODE=${nic_mode_str}"
+
+  rm -f ${tmp_file}
```

Comment:
No need for this line if we use TMP_DIR

Reply:
The only thing is that `rm -f ${tmp_file}` is already used in many places throughout the bfcfg code so I feel like it may be more confusing when reading through if that line doesn't exist only for a few functions. I could remove all of them, but I'm a little worried about causing regressions...

Does the following guarantee the file will be removed if called inside a function/subshell?
```
trap "rm -rf $TMP_DIR" EXIT
trap "rm -rf $TMP_DIR; exit 1" TERM INT
```
I tried to test it out and saw that the directory was still there after the function exited.

Reply:
The 'trap' command should guarantee the file to be removed I think. So only need to make sure the old tmp_file generated by one function not to be unused by another function incorrectly.

Reply:
Removed 'rm -f' for the new bf modes and pass settings dump functions and made sure they use a unique temp file name